open_prices/proofs/ml.py

```diff
@@ -947,17 +948,46 @@ def create_receipt_items_from_proof_prediction(
     proof: Proof, proof_prediction: ProofPrediction
 ) -> list[ReceiptItem]:
     """Create receipt items from a proof prediction containing receipt item
-    detections."""
+    detections.
+    Also looks up Prices with similar product name and location to try to
+    extract the product code.
 
+    :param proof: the Proof instance to associate the ReceiptItems with
+    :param proof_prediction: the ProofPrediction instance containing the
+        receipt items detections
+    :return: the list of ReceiptItem instances created
+    """
     if proof_prediction.model_name != common_google.GEMINI_MODEL_NAME:
         logger.error(
             "Proof prediction model %s is not a receipt extraction",
             proof_prediction.model_name,
         )
         return []
 
+    # For every predicted item product name, look up a corresponding Price
+    product_names = [
+        item.get("product_name")
+        for item in proof_prediction.data.get("items", [])
+        if item.get("product_name")
+    ]
+    matching_prices = Price.objects.filter(
+        product_name__in=product_names,
+        location=proof.location,
+    )
+    # Using the prices data, create a lookup table matching product names
+    # and product codes
+    product_lookup = {
```

Comment:
I would check here that we don't get 2 two different prices, and add a logger.warning (that will be logged by Sentry) in case we have.